### PR TITLE
Us193 rebased

### PR DESCRIPTION
**Pull request by dzadroga**
Completes US 193 and includes Tasks 194 to 196

Code completes:
- Design Turn Banner UI element (task 194)
- Bind to TurnStartedEvent (task 195)
- Animate transition on player change (task 196)

More testing will need to be completed later on in the project since it is hard to test the actual UI elements without a true turn being completed.

### CODE REVIEW

File: Assets/Scripts/UI/TurnBannerUI.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+using UnityEngine;
+
+public class TurnBannerUI
```
Comment (hdathert) [inline Assets/Scripts/UI/TurnBannerUI.cs:7]: What is this for?

File: Assets/Scripts/UI/TurnCyclerDirect.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+using UnityEngine;
+
+public class TurnCyclerDirect : MonoBehaviour
```
Comment (hdathert) [inline Assets/Scripts/UI/TurnCyclerDirect.cs]: I'm not sure what this does in the context of this UI. It seems to cycle the banner between players, and also activates it, but it seems like this behavior is better kept in the banner itself where it receives turn change events.

The TurnStartedEvent payload also includes the player id, so we can just get the player number from that anyway.

File: Assets/Scripts/UI/TurnBannerController.cs [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+        _shownPos = _rect.anchoredPosition;
+        _hiddenPos = _shownPos + new Vector2(0, -slideOffset);
+
+        canvasGroup.alpha = 0;
+        canvasGroup.interactable = false;
+        canvasGroup.blocksRaycasts = false;
+        _rect.anchoredPosition = _hiddenPos;
+
+        gameObject.SetActive(false);
```
Comment (hdathert) [inline Assets/Scripts/UI/TurnBannerController.cs]: If we set the game object to inactive, and in OnDisable we're severing the connection to event channels, then how is this meant to receive and react to events?

It seems like we should let this fully control its own state, and let it react via event channels instead- like by having the state of the Canvas stuff attached to this start at completely transparent, and then have this exist in space on its own. We don't need to fully disable it, since its meant to be reactive to events firing from other parts of our system.

Comment (hdathert) [reply Assets/Scripts/UI/TurnBannerController.cs]: While technically the events should still execute even when this is disabled, its certainly not exactly clear that is the case.

File: Assets/Scripts/UI/TurnBannerController.cs [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+    [Header("Events")]
+    [SerializeField] private TurnStartedEventChannel turnStartedChannel;
+
+    [Header("Animation Settings")]
+    [SerializeField] private float fadeDuration = 0.25f;
+    [SerializeField] private float slideOffset = 60f;
+    [SerializeField] private AnimationCurve ease = null;
+
+    private RectTransform _rect;
```
Comment (hdathert) [inline Assets/Scripts/UI/TurnBannerController.cs]: I believe the style guide for unity has private member variables as just the names, no underscore, for example:
```c#
private RectTransform rect;
private Vector2 shownPos;
```
etc...

**Review by hdathert — CHANGES_REQUESTED** [review]
Couple small things that I think need addressing.

File: Assets/Scripts/UI/TurnBannerUI.cs (hunk 1 above)
Comment (dzadroga) [reply Assets/Scripts/UI/TurnBannerUI.cs:7]: I created that file as a placeholder for UI integrations.  Right now since there is not much additional logic, the TurnBannerController handles the animation as well as the direct UI behavior.  In a later version, I thought we could wrap and expose public UI properties so that other systems can interact with the banner while still having a clean interface.  I added a comment explaining this now to help avoid confusion.  For now we can leave it empty and then start we introduce the UI abstraction layer assuming that is the direction we end up taking.

File: Assets/Scripts/UI/TurnCyclerDirect.cs (hunk 2 above)
Comment (dzadroga) [reply Assets/Scripts/UI/TurnCyclerDirect.cs]: Originally, I created that as a helper for PlayMode testing to manually cycle players and trigger the banner.  I decided to switch to EditMode tests that drive the banner directly and forgot to remove that file.  I deleted it from the project.

File: Assets/Scripts/UI/TurnBannerController.cs (hunk 3 above)
Comment (dzadroga) [reply Assets/Scripts/UI/TurnBannerController.cs]: I switched the banner to remain active and control visibility with the CanvasGroup.  This will keep the EventChannel subscriptions intact, makes the components clearly reactive to events, and removes any ambiguity for the disabled objects receiving events.

**Review by hdathert — APPROVED** [review]
# Code Review Checklist

**Completed for**: US #193 and Tasks #195 to #196 
**Code Review Completed By**: @hdathert 

Use this checklist to review new code changes before merging.

---

## Implementation
- [x] Does this code change accomplish what it is supposed to do?
- [x] Can the solution be simplified?
- [x] This change does **not** add unwanted compile-time or run-time dependencies.
- [x] The code is at the right abstraction level.
- [x] The code is modular enough.
- [x] The solution is adequate for maintainability, readability, performance, and security.
- [x] Similar functionality does **not** already exist in the codebase (or reuse is not appropriate).
- [x] No additional best practices or design patterns would substantially improve this code.

---

## Logic Errors and Bugs
- [x] No use cases where the code behaves incorrectly.
- [x] No inputs or external events that can break the code.
- [x] Code agrees with specifications.
- [x] No off-by-one errors.
- [x] No magic numbers (or they are well-named constants).
- [x] Proper use of global/large-scope variables (minimized and justified).

---

## Error Handling and Logging
- [x] Error handling is implemented correctly.
- [x] Error messages are user-friendly (where applicable).
- [x] Sufficient, meaningful log events exist for debugging (no sensitive data in logs).

---

## Dependencies
- [x] Documentation/configuration/README updates were made as needed.
- [x] No negative impact on other parts of the system or backward compatibility.

---

## Security and Data Privacy
- [x] No new security vulnerabilities introduced.
- [x] All user input is validated, sanitized, and escaped (XSS/SQL injection prevented).
- [x] Data from external APIs/libraries is handled safely and checked as needed.

---

## Performance
- [x] No significant performance regressions.
- [x] No obvious opportunities for significant performance gains left unaddressed.

---

## Ethics and Morality
- [x] Appropriate measures exist to prevent or report harassment/abuse (if applicable).

---

## Testing and Testability
- [x] Code is testable.
- [x] Automated tests have been added or updated.
- [x] Existing tests sufficiently cover the change (unit/integration/system).
- [x] Edge cases and boundary inputs are tested.

---

## Readability
- [x] Code is easy to understand.
- [x] Methods/functions are appropriately small and focused.
- [x] Names (functions/classes/variables) are clear and descriptive.
- [x] Code is in the correct file/folder/package.
- [x] Control flow is intuitive; data flow is clear.
- [x] Comments are clear, relevant, and up to date.
- [x] No commented-out or redundant code remains.
- [x] No overly dense one-liners; no overly long methods.

---

**Use this before submitting a pull request.**  
It helps ensure code quality, maintainability, and compliance with best practices.

*Based on awesome code review guides and MIT’s code review recommendations.*

> Looks good, thanks!

**Review by kescob0825 — COMMENTED** [review]
# Code Review Checklist

**Completed for**: US #193 and Tasks #194 to #196
**Code Review Completed By**: Karl Escobar

Use this checklist to review new code changes before merging.

---

## Implementation
- [x] Does this code change accomplish what it is supposed to do?
- [x] Can the solution be simplified?
- [x] This change does **not** add unwanted compile-time or run-time dependencies.
- [x] The code is at the right abstraction level.
- [x] The code is modular enough.
- [x] The solution is adequate for maintainability, readability, performance, and security.
- [x] Similar functionality does **not** already exist in the codebase (or reuse is not appropriate).
- [x] No additional best practices or design patterns would substantially improve this code.

---

## Logic Errors and Bugs
- [x] No use cases where the code behaves incorrectly.
- [x] No inputs or external events that can break the code.
- [x] Code agrees with specifications.
- [x] No off-by-one errors.
- [x] No magic numbers (or they are well-named constants).
- [x] Proper use of global/large-scope variables (minimized and justified).

---

## Error Handling and Logging
- [x] Error handling is implemented correctly.
- [x] Error messages are user-friendly (where applicable).
- [x] Sufficient, meaningful log events exist for debugging (no sensitive data in logs).

---

## Dependencies
- [x] Documentation/configuration/README updates were made as needed.
- [x] No negative impact on other parts of the system or backward compatibility.

---

## Security and Data Privacy
- [x] No new security vulnerabilities introduced.
- [x] All user input is validated, sanitized, and escaped (XSS/SQL injection prevented).
- [x] Data from external APIs/libraries is handled safely and checked as needed.

---

## Performance
- [x] No significant performance regressions.
- [x] No obvious opportunities for significant performance gains left unaddressed.

---

## Ethics and Morality
- [x] Appropriate measures exist to prevent or report harassment/abuse (if applicable).

---

## Testing and Testability
- [x] Code is testable.
- [x] Automated tests have been added or updated.
- [x] Existing tests sufficiently cover the change (unit/integration/system).
- [x] Edge cases and boundary inputs are tested.

---

## Readability
- [x] Code is easy to understand.
- [x] Methods/functions are appropriately small and focused.
- [x] Names (functions/classes/variables) are clear and descriptive.
- [x] Code is in the correct file/folder/package.
- [x] Control flow is intuitive; data flow is clear.
- [x] Comments are clear, relevant, and up to date.
- [x] No commented-out or redundant code remains.
- [x] No overly dense one-liners; no overly long methods.

---
